src/ServerManager.ts

```diff
@@ -3,19 +3,37 @@ import { verifyPeer } from './common/token';
 import MediaService from './MediaService';
 import { Peer } from './Peer';
 import Room from './Room';
+import ManagementService from './ManagementService';
 
 const logger = new Logger('ServerManager');
 
+interface ServerManagerOptions {
+	mediaService: MediaService;
+	peers: Map<string, Peer>;
+	rooms: Map<string, Room>;
+	managedPeers: Map<string, Peer>;
+	managedRooms: Map<string, Room>;
+	managementService: ManagementService;
+}
+
 export default class ServerManager {
 	public closed = false;
-	public peers = new Map<string, Peer>();
-	public rooms = new Map<string, Room>();
+	public peers: Map<string, Peer>;
+	public rooms: Map<string, Room>;
+	public managedRooms = new Map<string, Room>(); // Mapped by ID from management service
```

Comment:
We create the maps for managedRooms and managedPeers here, but not for peers and rooms.
All four are created in server.ts anyway, and replaced in ServerManager constructor.

Perhaps be consistent here and not create the Map objects for managedRooms and managedPeers?